### ISSUE TITLE
=> AP: `contentMap` values should be HTML

### ISSUE THREAD

**Issue by Tamschi** [snarfed/bridgy-fed#1609]
Current example with plaintext in #1607. It should have the same semantics as `content`: https://github.com/snarfed/bridgy-fed/issues/1607#issue-2723546263

The properties are in fact mapped to the same key in `https://www.w3.org/ns/activitystreams`, just with different transforms:

```json
"content": "as:content",
"contentMap": {
    "@id":"as:content",
    "@container":"@language"
},
```

(This is valid, they just become an array after expansion like this:

```json
"https://www.w3.org/ns/activitystreams#content": [
  {
    "@value": "<p>OMG OMG OMG we're able to bridge our threads accounts into Blue sky via federation <a class=\"mention\" href=\"https://ap.brid.gy/ap.brid.gy\">@ap.brid.gy</a>  & <a class=\"mention\" href=\"https://bsky.app/profile/snarfed.org\">@snarfed.org</a><br><br>RE: <a href=\"https://bsky.app/profile/did:plc:4ghwklnin2mvmam4ditdwv2o/post/3lclod7ptodd2\">https://bsky.app/profile/did:plc:4ghwklnin2mvmam4ditdwv2o/post/3lclod7ptodd2</a></p>"
  },
  {
    "@value": "OMG OMG OMG we're able to bridge our threads accounts into Blue sky via federation @ap.brid.gy  & @snarfed.org",
    "@language": "en"
  }
],
```

(via https://json-ld.org/playground/)

**Comment by Tamschi** [comment]
Do Firefish and Catodon allow toggling the post display language if there's a `contentMap`?

**Comment by Tamschi** [comment]
I had a look at [catodon's code](https://codeberg.org/catodon/catodon/src/commit/dbd2d1ce4df617d414a6b108b7d8bfb4fe815bd7/packages/backend/src/remote/activitypub/models/note.ts#L351-L376). In order of decreasing priority it uses:

- `source.content` iff `source.mediaType` is `"text/x.misskeymarkdown"`, language-tagged[^1] with the first key in `contentMap`.
- The first entry in `contentMap`.
- Nothing if `contentMap` is present but empty.[^2]
- `_misskey_content`.
- `content`

So no, no mutli-language support, it just picks the content a bit differently.
It appears to use MFM as native storage format, so any HTML is converted into that representation.

[^1]: Language-tagging is subject to an allow-list of supported languages. If not supported, the information is discarded entirely, I think.
[^2]: This directly mismatches JSON-LD processing rules, I think. Oh well.

**Comment by panosda** [comment]
Hey, sorry for not being more helpful, but I'm not a dev. I just asked in Iceshrimp and a dev told me that bridgy fed doesn't include the correct link in contentmap, and that this is the reason links don't work for us. Do you think this could be fixed on your side? Thanks!

**Comment by Tamschi** [comment]
Yes, this is squarely a bug on our side. Catodon's behaviour is correct here, just a little unusual.

**Comment by snarfed** [comment]
Done! Will be deployed in ~10m.
